script.js

```diff
@@ -35,21 +35,37 @@ class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      user: null };
+		user: null,
+		toast: null};
 
   }
   signIn(email, password) {
-    // sign in APIs here
-    this.setState({
-      user: {
-        email,
-        password } });
-
-
+		axios.post('http://localhost:3000/login',{
+			username: email,
+			password: password
+		})
+		.then((response) => {
+			if(response.status === 201){this.setState({
+				user: 
+				{
+					email,
+					password 
+				}, 
+				toast: null
+			});
+			}
+		}, (error) => {
+			if(error.response.status === 403) {this.setState( {
```

Comment:
Similarly to the above, if the server were to throw an error (e.g. 500 error), or if a different type of error status were returned (e.g. 400 Bad Request), this code would fail silently and do nothing.

Reply:
method is now generalized so any error will be outputted as toast notification using the server's response as the text